aries-site/src/examples/templates/content-layouts/ContentDrivenLayout.js

```diff
@@ -0,0 +1,110 @@
+import PropTypes from 'prop-types';
+import {
+  Box,
+  Button,
+  Form,
+  FormField,
+  Notification,
+  Paragraph,
+  Text,
+  TextInput,
+} from 'grommet';
+
+import { ModalBody, ModalDialog, ModalFooter } from '../ModalDialog';
+
+const path = '/products/data/47287498729/collections/KCHDvfcByKvvjymNheg';
+const parts = path.split('/').map(part => `/${part}`);
+const matchPath = [
+  ...parts.map(part => ({
+    regexp: new RegExp(part),
+    message: `Location paths do not match. Missing '${part}'.`,
+    status: 'error',
+  })),
+  function validate(value) {
+    if (value !== path) {
+      return {
+        message: `Location paths do not match. 
+Expected: ${path}
+Received: ${value}`,
+        status: 'error',
+      };
+    }
+    return undefined;
+  },
+];
+
+const defaultValues = { deletionPath: '' };
+
+export const ContentDrivenLayout = ({ containerRef }) => {
+  // containerRef is for demonstration purposes on this site. Most
+  // implementations should likely remove.
+  const [showModal, setShowModal] = useState(true);
+  const [value, setValue] = useState(defaultValues);
+  const onClose = () => {
+    setShowModal(false);
+    setValue(defaultValues);
+  };
+  const onChange = nextValue => {
+    setValue(nextValue);
+  };
+
+  return (
+    <Box align="center" justify="center" fill>
+      <Button
+        primary
+        label="Display Modal"
+        onClick={() => setShowModal(true)}
+      />
+      {showModal ? (
+        <ModalDialog
+          title="Delete Data"
+          onEsc={onClose}
+          target={containerRef && containerRef.current}
+        >
+          <Form value={value} onChange={onChange} validate="blur">
+            <Box gap="medium">
+              <ModalBody gap="small">
+                <Notification
+                  status="critical"
+                  message="This action cannot be undone."
+                />
+                <>
+                  <Paragraph margin="none">
+                    This will permanently delete all data, including nested
+                    data, located at:
+                  </Paragraph>
+                  <Text weight="bold">{path}</Text>
+                </>
+                <FormField
+                  htmlFor="deletionPath"
+                  name="deletionPath"
+                  label={
+                    <>
+                      <Text>
+                        To confirm, please type: <br />
+                        {path}
+                      </Text>
+                    </>
+                  }
+                  validate={matchPath}
+                >
+                  <TextInput id="deletionPath" name="deletionPath" />
+                </FormField>
+              </ModalBody>
+              <ModalFooter>
+                <></>
+                <Box direction="row" gap="xsmall">
+                  <Button label="Cancel" onClick={onClose} />
+                  <Button primary label="Delete" onClick={() => {}} />
```

Comment:
```suggestion
                  <Button primary label="Delete" onClick={() => {}} type="submit" />
```
Do we need this? Also, the first time I click Delete with no content typed, I don't get any form error.

Reply:
added